returnn/search.py

```diff
@@ -14,6 +14,178 @@
 import recipe.i6_core.util as util
 
 
+class ReturnnSearchJob(Job):
+    """
+    Given a model checkpoint, run search task with RETURNN
+    """
+
+    def __init__(
+        self,
+        search_data,
+        model_checkpoints,
+        epoch,
+        returnn_config,
+        *,
+        log_verbosity=3,
+        device="gpu",
+        time_rqmt=4,
+        mem_rqmt=4,
+        cpu_rqmt=2,
+        returnn_python_exe=None,
+        returnn_root=None,
+    ):
+        """
+        :param dict[str] search_data: dataset used for search
+        :param dict[int, Checkpoint] model_checkpoints: Contains checkpoints pointing to TF models.
+            see `ReturnnTrainingJob`.
+        :param tk.Variable|int epoch: checkpoint's epoch to load for search
+        :param ReturnnConfig returnn_config: object representing RETURNN config
+        :param int log_verbosity: RETURNN log verbosity
+        :param str device: RETURNN device, cpu or gpu
+        :param float|int time_rqmt: job time requirement in hours
+        :param float|int mem_rqmt: job memory requirement in GB
+        :param float|int cpu_rqmt: job cpu requirement in GB
+        :param tk.Path|str|None returnn_python_exe: path to the RETURNN executable (python binary or launch script)
+        :param tk.Path|str|None returnn_root: path to the RETURNN src folder
+        """
+        self.search_data = search_data
+
+        self.model_checkpoint = model_checkpoints[
+            epoch.get() if isinstance(epoch, tk.Variable) else epoch
+        ]
+
+        self.returnn_python_exe = (
+            returnn_python_exe
+            if returnn_python_exe is not None
+            else gs.RETURNN_PYTHON_EXE
+        )
+
+        self.returnn_root = (
+            returnn_root if returnn_root is not None else gs.RETURNN_ROOT
+        )
+
+        self.out_search_file = self.output_path("search_out")
+
+        self.returnn_config = self.create_returnn_config(
```

Comment:
This should not be in the middle of the output definitions I think...

Reply:
this won't work since `self.out_search_file` is used in `self.create_returnn_config`

Reply:
But you can put it after it, right?

Reply:
ah yeah to fix the order of output paths. ok